src/input/plugins/CdioParanoiaCDID.cxx

```diff
@@ -0,0 +1,357 @@
+#define CDROM_LEADOUT           0xAA             /* leadout track */
+#define CDROMREADTOCHDR         CDIOREADTOCHEADER
+#define CDROMREADTOCENTRY       CDIOREADTOCENTRY
+#define cdrom_tochdr            ioc_toc_header
+#define cdth_trk0               starting_track
+#define cdth_trk1               ending_track
+#define cdrom_tocentry          ioc_read_toc_single_entry
+#define cdte_track              track
+#define cdte_format             address_format
+#define cdte_track_address      entry.addr.lba
+#define DEVICE_NAME             "/dev/cdrom"
+
+#elif defined(__OpenBSD__) || defined(__NetBSD__)
+#include <sys/cdio.h>
+#define CDROM_LBA               CD_LBA_FORMAT    /* first frame is 0 */
+#define CD_MSF_OFFSET           150              /* MSF offset of first frame */
+#define CD_FRAMES               75               /* per second */
+#define CDROM_LEADOUT           0xAA             /* leadout track */
+#define CDROMREADTOCHDR         CDIOREADTOCHEADER
+#define cdrom_tochdr            ioc_toc_header
+#define cdth_trk0               starting_track
+#define cdth_trk1               ending_track
+#define cdrom_tocentry          cd_toc_entry
+#define cdte_track              track
+#define cdte_track_address      addr.lba
+#define DEVICE_NAME             "/dev/cd0a"
+
+#elif defined(__APPLE__)
+#include <sys/types.h>
+#include <IOKit/storage/IOCDTypes.h>
+#include <IOKit/storage/IOCDMediaBSDClient.h>
+#define CD_FRAMES               75               /* per second */
+#define CD_MSF_OFFSET           150              /* MSF offset of first frame */
+#define cdrom_tochdr            CDDiscInfo
+#define cdth_trk0               numberOfFirstTrack
+/* NOTE: Judging by the name here, we might have to do this:
+ * hdr.lastTrackNumberInLastSessionMSB << 8 *
+ * sizeof(hdr.lastTrackNumberInLastSessionLSB)
+ * | hdr.lastTrackNumberInLastSessionLSB; */
+#define cdth_trk1               lastTrackNumberInLastSessionLSB
+#define cdrom_tocentry          CDTrackInfo
+#define cdte_track_address      trackStartAddress
+#define DEVICE_NAME             "/dev/rdisk1"
+
+#elif defined(__sgi)
+#include <dmedia/cdaudio.h>
+#define CD_FRAMES               75               /* per second */
+#define CD_MSF_OFFSET           150              /* MSF offset of first frame */
+#define cdrom_tochdr            CDSTATUS
+#define cdth_trk0               first
+#define cdth_trk1               last
+#define close                   CDclose
+struct cdrom_tocentry
+{
+	int cdte_track_address;
+};
+#define DEVICE_NAME             "/dev/scsi/sc0d4l0"
+#else
+#error "Your OS isn't supported yet."
+#endif  /* os selection */
+}
+
+static char*
+makeMusicBrainzIdWith(int firstTrack, int lastTrack, int leadIn, std::vector<int> frameOffsets)
+{
+	struct AVSHA* sha1 = av_sha_alloc();
+	const int numBitsSha1 = 160;
+	const int digestSize = numBitsSha1 / 8;
+	uint8_t digest[digestSize];
+	const int tempSize = 32;
+	char temp[tempSize];
```

Comment:
This is a variable-length array which is illegal in C++ (and even deprecated in C).